### PR TITLE
GTEST: Use TMPDIR for the location of temporary files if set

### PR DESCRIPTION
**Pull request by MattBBaker**
On some systems /tmp isn't the best option or usable. Like Cray clusters. Remove the hard coded /tmp in gtest to use the unix standard TMPDIR env variable if it is present.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4621/ for details (Mellanox internal link).

File: test/gtest/ucs/test_log.cc [hunk 1]
```diff
@@ -13,6 +13,8 @@ class log_test : public ucs::test {
 public:
     virtual void init() {
         char ucs_log_spec[70];
+        char *default_tmp_dir = "/tmp";
```
Comment (shamisp) [inline test/gtest/ucs/test_log.cc]: cont char *

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4635/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@MattBBaker coverity error - because the value returned from getenv() influences the parameter passed to unlink()
can silence it by adding the comment `/* coverity[tainted_string] */` one the line before the call to unlink()

**Comment by MattBBaker** [comment]
Hopefully that silences coverity.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4648/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@MattBBaker pls format the commit messages (including the updates / fixes) according to code style

**Pull request merged**